src/bitcoin_adaptor.rs

```diff
@@ -201,9 +198,9 @@ mod test {
 
         let genesis = genesis_block(Network::Bitcoin);
         // store the genesist block
-        bdb.put_hash_keyed(&genesis).unwrap();
+        bdb.put_hash_keyed(&genesis.bitcoin_hash().as_hash(), &genesis).unwrap();
         // find it
-        if let Some((_, block)) = bdb.get_hash_keyed::<Block>(&genesis.bitcoin_hash()).unwrap() {
+        if let Some((_, block)) = bdb.get_hash_keyed::<Block>(&genesis.bitcoin_hash().as_hash()).unwrap() {
```

Comment:
Maybe use `block_hash` instead?

Reply:
block_hash is only available on master at the moment, got to wait for the next rust-bitcoin release